Packages miss backwards-compatibility link of testing-related files

It seems that our binary packages, at least for zeek-6.0.0 and debian-11 miss a softlink of `<PREFIX>/share/btest/data` to `<PREFIX>/share/zeek/tests`. We have an explicit install rule for that, https://github.com/zeek/zeek/blob/4b71b7d05c7fad7bd2c2619471a836a142d8edf2/testing/CMakeLists.txt#L12-L18

This leads to packages relying on support files like `random.seed` or `Scripts/canonify-zeek-log` to be in this path to fail, e.g., a user reported that he was unable to install `zeek/spicy-ldap` in his system with Zeek from packages while our CI was ✅ (our CI uses Docker images which have these files).

We should either install this softlink in packages as well, or alternatively get rid of it completely in our CMake setup so both installations methods are more similar.

@bbannier: found the reason why this is not part of binary packages.

If one uses `make install DESTIR=[staging directory]` the symlink is not created. Instead, the following output is generated:

```
[ 3977s] CMake Error: failed to create symbolic link '/opt/zeek/share/zeek/tests': No such file or directory
```

If we create a relative link as a symlink, instead of the absolute link, this will ba packaged.

--- a/testing/CMakeLists.txt
+++ b/testing/CMakeLists.txt
@@ -14,7 +14,7 @@ install(
     CODE "execute_process( \
     COMMAND ${CMAKE_COMMAND} -E create_symlink \
     ${ZEEK_CONFIG_BTEST_TOOLS_DIR}/data \
-    ${CMAKE_INSTALL_PREFIX}/share/zeek/tests \
+    \$ENV{DESTDIR}/${CMAKE_INSTALL_PREFIX}/share/zeek/tests \
     )")
 
 install(DIRECTORY scripts/spicy/ DESTINATION ${ZEEK_CONFIG_BTEST_TOOLS_DIR}/data/Scripts

